src/server/game/DungeonFinding/LFGMgr.cpp

```diff
@@ -1603,21 +1611,44 @@ namespace lfg
 
         _SaveToDB(gguid);
 
+        // Select a player inside to be teleported to
+        WorldLocation const* teleportLocation = nullptr;
+        for (GroupReference* itr = grp->GetFirstMember(); itr != nullptr; itr = itr->next())
+        {
+            Player* plr = itr->GetSource();
+            if (plr && plr->GetMapId() == uint32(dungeon->map) && !proposal.isNew)
+            {
+                teleportLocation = plr;
+                break;
+            }
+        }
+
         bool randomDungeon = false;
         // Teleport Player
         for (LfgGuidList::const_iterator it = playersToTeleport.begin(); it != playersToTeleport.end(); ++it)
             if (Player* player = ObjectAccessor::FindPlayer(*it))
             {
                 if (player->GetGroup() != grp) // pussywizard: could not add because group was full (some shitness happened)
                     continue;
+
+                if (player->GetMapId() == uint32(dungeon->map))
```

Comment:
I see that further down you use static_cast and here not. What is the difference between the different cases?

Reply:
None.  `uint32` cast is c style.

Reply:
yea, I just want to learn why here you use `player->GetMapId() == uint32(dungeon->map)` and in a line below is `player->GetMapId() == static_cast<uint32>(dungeon->map)`

Reply:
Just laziness :P
Both doing the same.